tests/models/prism/test_tokenization_prism.py

```diff
@@ -0,0 +1,203 @@
+    pass
+
+EN_CODE = 37
+FR_CODE = 71
+
+
+@require_sentencepiece
+class PrismTokenizationTest(TokenizerTesterMixin, unittest.TestCase):
+    from_pretrained_id = "dariast/prism"
+    tokenizer_class = PrismTokenizer
+    test_rust_tokenizer = False
+    test_seq2seq = False
+    test_sentencepiece = True
+
+    def setUp(self):
+        super().setUp()
+
+        vocab = ["</s>", "<unk>", "▁This", "▁is", "▁a", "▁t", "est", "\u0120", "<pad>", "<en>"]
+        vocab_tokens = dict(zip(vocab, range(len(vocab))))
+        save_dir = Path(self.tmpdirname)
+        save_json(vocab_tokens, save_dir / VOCAB_FILES_NAMES["vocab_file"])
+        if not (save_dir / VOCAB_FILES_NAMES["spm_file"]).exists():
+            copyfile(SAMPLE_SP, save_dir / VOCAB_FILES_NAMES["spm_file"])
+
+        tokenizer = PrismTokenizer.from_pretrained(self.tmpdirname)
+        tokenizer.save_pretrained(self.tmpdirname)
+
+    def get_tokenizer(self, **kwargs):
+        return PrismTokenizer.from_pretrained(self.tmpdirname, **kwargs)
+
+    def test_convert_token_and_id(self):
+        """Test ``_convert_token_to_id`` and ``_convert_id_to_token``."""
+        token = "</s>"
+        token_id = 0
+
+        self.assertEqual(self.get_tokenizer()._convert_token_to_id(token), token_id)
+        self.assertEqual(self.get_tokenizer()._convert_id_to_token(token_id), token)
+
+    def test_get_vocab(self):
+        tokenizer = self.get_tokenizer()
+        vocab_keys = list(tokenizer.get_vocab().keys())
+
+        self.assertEqual(vocab_keys[0], "</s>")
+        self.assertEqual(vocab_keys[1], "<unk>")
+        self.assertEqual(vocab_keys[-1], "<s>")
+
+    def test_full_tokenizer(self):
+        tokenizer = self.get_tokenizer()
+
+        tokens = tokenizer.tokenize("This is a test")
+        self.assertListEqual(tokens, ["▁This", "▁is", "▁a", "▁t", "est"])
+
+        self.assertListEqual(
+            tokenizer.convert_tokens_to_ids(tokens),
+            [2, 3, 4, 5, 6],
+        )
+
+        back_tokens = tokenizer.convert_ids_to_tokens([2, 3, 4, 5, 6])
+        self.assertListEqual(back_tokens, ["▁This", "▁is", "▁a", "▁t", "est"])
+
+        text = tokenizer.convert_tokens_to_string(tokens)
+        self.assertEqual(text, "This is a test")
+
+
+@require_torch
+@require_sentencepiece
+@require_tokenizers
+class PrismTokenizerIntegrationTest(unittest.TestCase):
+    checkpoint_name = "dariast/prism"
+    src_text = ["Hi world.", "This is a Test.", "Some of my Best Friends are Linguists."]
+    tgt_text = ["<fr> Hé, monde!", "<fr> C'est un test.", "<fr> Certains de mes meilleurs amis sont linguistes."]
+
+    expected_src_tokens = [37, 5050, 21, 1951, 13934, 33789, 7, 269, 11348, 983, 9393, 6, 2]
+
+    @classmethod
+    def setUpClass(cls):
+        cls.tokenizer: PrismTokenizer = PrismTokenizer.from_pretrained(
+            cls.checkpoint_name, src_lang="en", tgt_lang="fr"
+        )
+        cls.pad_token_id = 1
+        return cls
+
+    def test_language_codes(self):
```

Comment:
The test is added with a minor change. 
```self.assertIn(FR_CODE, self.tokenizer.all_special_ids)``` is not applicable to the Prism model because it doesn't include target language IDs in the list of special tokens.

Reply:
I assume with the fix you made, now it should be possible to add the line back.

Reply:
Added the line ```self.assertIn(FR_CODE, self.tokenizer.all_special_ids)```  and tested, all tests pass.